### DISCUSSION TITLE
Prefixing

### DISCUSSION THREAD

**Discussion by nlemoine** [BrianHenryIE/strauss#10] [Q&A]
Sorry for the dumb question, I'm currently testing `strauss`. Is it taking care of replacing dependencies in my project/package? Or do I have to prefix them myself?

I get my vendor directory correctly prefixed but my `src` directory (`{""psr-4": {"MyOwnNs\\": "src"}}`) still uses the original dependency namespaces:

```php
use GuzzleHttp\Client;
```

Thanks!

**Comment by BrianHenryIE** [comment]
Hey,

It does not change your existing code, only the dependencies that it copies, so yes, you have to prefix them yourself.

I do think it's a good idea that it would do that. It shouldn't be too difficult to implement, but I haven't had the time to think it through. Since, as you say, the autoload key is telling us where to look, the replacements could easily be made. And how they are performed already would mean they would only happen once, no matter how many times Strauss is run.

Similarly, I think it might eventually be possible to prefix the namespaces and classnames of files inside `/vendor/` without bothering to copy them at all.

As you probably see, I only published this fork a week ago. I expect there'll be a few bugs to iron out, then as I fix them and mull things over, the best way to implement these ideas will fall into place.

Thanks for the PR. And this post – I think this discussion group has great potential for hashing out ideas.

**Reply by nlemoine** [reply]
Thanks for the prompt reply and explanation!
